### PR TITLE
fix: disabled feature walkthroughs behind a flag

### LINKED ISSUE #30339: [Bug]: Query data fast popup visible everywhere
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The Query data fast popup appears in other places then the query page. I remember seeing it on the canvas too, but today the user saw it in the apps list page after visiting the query page.

image

### Steps To Reproduce

Go to the query page for the first time then click on the appsmith logo

### Public Sample App

_No response_

### Environment

Release

### Issue video log

_No response_

### Version

1.9.59

### PR DESCRIPTION
**Pull request by sneha122**
## Description

This PR removes the feature walkthroughs that were added as part of one of the activation experiments in Q3. Although these walkthroughs were useful in guiding users, they were appearing in untimely and incorrect fashion. 

#### PR fixes following issue(s)
Fixes #30339 
- Bug fix (non-breaking change which fixes an issue)
>
>
>
- [x] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Introduced a feature flag to toggle the feature walkthrough on or off.
	- Added the ability to force display the feature walkthrough regardless of the feature flag setting.

### CODE REVIEW

File: app/client/src/components/featureWalkthrough/index.tsx [hunk 1]
```diff
@@ -27,17 +36,19 @@ export default function Walkthrough({ children }: any) {
...
+      if (!alreadyExists) {
+        const _value = Array.isArray(value) ? [...value] : [value];
+        if (prioritize) {
+          // Get ahead of the queue
+          setFeature((e) => [..._value, ...e]);
+        } else {
+          setFeature((e) => [...e, ..._value]);
+        }
       }
+      updateActiveWalkthrough();
```
[the remark this answers is not in the document]
Comment (sneha122) [reply app/client/src/components/featureWalkthrough/index.tsx:50]: If the flag is set to true, we want to disable the walkthroughs, if the flag is false, we want to keep showing the walkthroughs, hope this helps in clearing the confusion

File: app/client/src/components/featureWalkthrough/walkthroughContext.tsx [hunk 2]
```diff
@@ -67,6 +67,9 @@ export interface FeatureParams {
   dismissOnOverlayClick?: boolean;
   // execute function just before showing walkthrough highlight
   runBeforeWalkthrough?: () => void;
+  // If we want to force the display of walkthrough, set this prop to true
+  // If set to true, will override the feature flag as well
+  forceExecution?: boolean;
```
Comment (AmanAgarwal041) [inline app/client/src/components/featureWalkthrough/walkthroughContext.tsx:72]: This parameter is defined but not used anywhere in the code to override the display of forceExecution. cc @sneha122

File: app/client/src/components/featureWalkthrough/index.tsx [hunk 3]
```diff
@@ -8,6 +8,15 @@ import { useLocation } from "react-router-dom";
 import AnalyticsUtil from "utils/AnalyticsUtil";
 import { isElementVisible } from "./utils";
 import { hideIndicator } from "components/utils/Indicator";
+import store from "store";
+import { FEATURE_FLAG } from "@appsmith/entities/FeatureFlag";
+import { selectFeatureFlagCheck } from "@appsmith/selectors/featureFlagsSelectors";
+
+const state = store.getState();
+const isTurnOffWalkthroughFlagSet = selectFeatureFlagCheck(
```
Comment (AmanAgarwal041) [inline app/client/src/components/featureWalkthrough/index.tsx:16]: Was selector function not working inside the component ? Confirming this as we are getting the overall state.

File: app/client/src/components/featureWalkthrough/walkthroughContext.tsx (hunk 2 above)
Comment (sneha122) [reply app/client/src/components/featureWalkthrough/walkthroughContext.tsx:72]: Let me add that, I thought we could have added that later but let me handle that in this PR itself.

File: app/client/src/components/featureWalkthrough/index.tsx (hunk 3 above)
Comment (sneha122) [reply app/client/src/components/featureWalkthrough/index.tsx:16]: I thought this is a ts utils file, let me try with selector

**Review by AmanAgarwal041 — APPROVED** [review]
lgtm

**Pull request merged**